### PR TITLE
CORE-15006. [RDBSSLIB][RXCE] Work around compiling as NT6+

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Addendum to r75398 (f8e1661) and r75441 (0eacafa).
@HeisSpiter

JIRA issue: [CORE-15006](https://jira.reactos.org/browse/CORE-15006)

### CODE REVIEW

**Comment by SergeGautherie** [comment]
> Starting with NT6, RBDSS, RXCE got totally rewritten to change the way they work.

Thanks for the explanation.

> Furthermore, this is a driver, you shouldn't check against WIN32 version, it's not related.

I simply adapted what is used in `ddk/*.h`: I don't know what else to use.

For what it's worth, they now continue to compile fine.
I now added `NFS`, to fix:
```
.../drivers/filesystems/nfs/nfs41_driver.c: In function 'nfs41_DevFcbXXXControlFile':
.../drivers/filesystems/nfs/nfs41_driver.c:2414:25: error: implicit declaration of function 'RxAcquirePrefixTableLockExclusive' [-Werror=implicit-function-declaration]
.../drivers/filesystems/nfs/nfs41_driver.c:2466:25: error: implicit declaration of function 'RxReleasePrefixTableLock' [-Werror=implicit-function-declaration]
```

> I'd rather you force NT5 build for this component through CMake.

`*_definitions( _WIN32_WINNT )`? Is it possible to have a condition around them?

File: drivers/filesystems/nfs/nfs41_driver.c [hunk 1]
```diff
@@ -19,6 +19,12 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA
  */
 
+/* NFSv4.1 relies on some NT5- functions */
```
Comment (HeisSpiter) [inline drivers/filesystems/nfs/nfs41_driver.c]: That would deserve a true explanation. I had to backport it to NT5, as it's initially written for Windows 7. So...

**Comment by HeisSpiter** [comment]
See: https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/kernel32_vista/CMakeLists.txt;h=4f1f09fb5ddf852cc71d12658fb198d007c9901f;hb=HEAD

File: sdk/lib/drivers/rdbsslib/rdbss.c [hunk 2]
```diff
@@ -26,6 +26,12 @@
 
 /* INCLUDES *****************************************************************/
 
+/* Starting with NT6, RDBSS got totally rewritten to change the way it works */
+#if (_WIN32_WINNT >= _WIN32_WINNT_VISTA)
+#undef _WIN32_WINNT
+#define _WIN32_WINNT _WIN32_WINNT_WS03
```
Comment (HeisSpiter) [inline sdk/lib/drivers/rdbsslib/rdbss.c]: Once again, WIN32 is no point in the kernel.

Comment (HBelusca) [reply sdk/lib/drivers/rdbsslib/rdbss.c]: The correct check is checking `NTDDI_VERSION` against `NTDDI_WS03` or `NTDDI_LONGHORN` (or `NTDDI_VISTA` for that matter).

File: drivers/filesystems/nfs/nfs41_driver.c (hunk 1 above)
Comment (SergeGautherie) [reply drivers/filesystems/nfs/nfs41_driver.c]: Updated.

**Comment by SergeGautherie** [comment]
> CMakeLists.txt

Done.

**Comment by HeisSpiter** [comment]
If the point is just about adding comments in CMakeLists.txt, this won't be merged to master.

**Comment by HeisSpiter** [comment]
My upper point still stands. Unless you have other plans, I'll just close the pull request.

**Comment by SergeGautherie** [comment]
> If the point is just about adding comments in CMakeLists.txt, this won't be merged to master.

Why?

> Unless you have other plans

Enabled it by default then: overkill, but what else?

**Comment by HeisSpiter** [comment]
We have no plans on NT6+ for now. If that's your playground, keep it local.

**Pull request closed without merging**